cs3api4lab/api/share_api_facade.py

```diff
@@ -0,0 +1,187 @@
+import cs3.ocm.provider.v1beta1.provider_api_pb2_grpc as ocm_provider_api_grpc
+import mimetypes
+
+from datetime import datetime
+from IPython.utils import tz
+from cs3api4lab.auth.authenticator import Auth
+from cs3api4lab.api.cs3_file_api import Cs3FileApi
+from cs3api4lab.common.strings import *
+from cs3api4lab.config.config_manager import Cs3ConfigManager
+from cs3api4lab.auth.channel_connector import ChannelConnector
+from cs3api4lab.api.cs3_user_api import Cs3UserApi
+
+from cs3api4lab.api.cs3_share_api import Cs3ShareApi
+from cs3api4lab.api.cs3_ocm_share_api import Cs3OcmShareApi
+
+import urllib.parse
+from cs3api4lab.api.share_utils import ShareUtils
+
+class ShareAPIFacade:
+    date_fmt = '%Y-%m-%dT%H:%M:%SZ'
+    TYPE_FILE = 1
+    TYPE_DIRECTORY = 2
+
+    def __init__(self, log):
+        self.log = log
+        self.config = Cs3ConfigManager().get_config()
+        self.auth = Auth.get_authenticator(config=self.config, log=self.log)
+        self.file_api = Cs3FileApi(log)
+
+        channel = ChannelConnector().get_channel()
+        self.provider_api = ocm_provider_api_grpc.ProviderAPIStub(channel)
+        self.user_api = Cs3UserApi(log)
+
+        self.share_api = Cs3ShareApi(log)
+        self.ocm_share_api = Cs3OcmShareApi(log)
+        return
+
+    def create(self, endpoint, file_path, opaque_id, idp, role=Role.EDITOR, grantee_type=Grantee.USER, reshare=True):
+        if self._is_ocm_user(opaque_id, idp):
+            return self.ocm_share_api.create(opaque_id, idp, idp, endpoint, file_path, grantee_type, role, reshare)
+        else:
+            return self.share_api.create(endpoint, file_path, opaque_id, idp, role, grantee_type)
+
+    def update_share(self, **kwargs):
+        if 'role' in kwargs:
```

Comment:
So the sharing API and OCM sharing API are not consistent and different parameters are used. In case of update 'role' parameter is the only one possible.

Reply:
I meant to put some comment explaining it. It will be easier in the future.

Reply:
Hi, so could you explain what comments do you need and where? I added some comments in the code

Reply:
I think it's enough as is now.